zebra/irdp_main.c

```diff
@@ -234,7 +234,8 @@ int irdp_send_thread(struct thread *t_advert)
       }
 
   tmp = irdp->MaxAdvertInterval-irdp->MinAdvertInterval;
-  timer =  (random () % tmp ) + 1;
+  assert (tmp > 0);
+  timer = (random () % tmp) + 1;
   timer = irdp->MinAdvertInterval + timer;
```

Comment:
that or the author wanted to make sure that we never got `minadvertinterval` as the timer value, in case `random()` returns `minadvertinterval` causing the remainder to be zero...anyway me and Donald discussed this as my original interpretation of the RFC was indeed that it was permissible for `minadvertinterval == maxadvertinterval`, however that seems a bit inconsistent because the RFC species a lower bound of `3` for the minimum but a lower bound of `4` for the maximum. Wouldn't make sense to allow them to be equal with the one exception that you can't enter `min = 3` and `max = 3` but you can enter `min = 4` and `max = 4` etc...